### PR TITLE
Patch CVE-2023-27533 in cmake

### PR DESCRIPTION
**Pull request by suresh-thelkar**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Patch CVE-2023-27533 in cmake.
###### Change Log  

- CVE-2023-27533

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- NA

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-27533

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

**Review by PawelWMS — APPROVED** [review]

**Review by mbykhovtsev-ms — APPROVED** [review]

**Comment by suresh-thelkar** [comment]
Thank you @PawelWMS and @mbykhovtsev-ms for the reviews.

File: SPECS/cmake/cmake.spec [hunk 1]
```diff
@@ -21,6 +21,7 @@ Patch6:         CVE-2023-38545.patch
 Patch7:         CVE-2023-38546.patch
 Patch8:         cve-2023-44487.patch
 Patch9:         CVE-2023-28320.patch
+Patch10:        CVE-2023-27533.patch
```
Comment (jslobodzian) [inline SPECS/cmake/cmake.spec:24]: What about CVE-2023-27534?

**Comment by suresh-thelkar** [comment]
Closing this PR as these changes are taken into the PR #10484

File: SPECS/cmake/cmake.spec (hunk 1 above)
Comment (suresh-thelkar) [reply SPECS/cmake/cmake.spec:24]: Thanks @jslobodzian. I have taken this PR changes into #10484 and hence closing this PR.

**Pull request closed without merging**